File: Definitions/Game_Definitions/classes/WallScreen.lua
---@meta
---@diagnostic disable

---@class WallScreen : TV
---@field movementPattern SMovementPattern
---@field factOnFullyOpened CName
---@field objectMover ObjectMoverComponent
WallScreen = {}

---@return WallScreen
function WallScreen.new() return end

---@param props table
---@return WallScreen
function WallScreen.new(props) return end

---@param movementStatus ObjectMoverStatus
---@return Bool
function WallScreen:OnMovementFinished(movementStatus) return end

---@param ri entEntityRequestComponentsInterface
---@return Bool
function WallScreen:OnRequestComponents(ri) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function WallScreen:OnTakeControl(ri) return end

---@param evt ToggleShow
---@return Bool
function WallScreen:OnToggleSecureShow(evt) return end

---@return WallScreenController
function WallScreen:GetController() return end

---@return WallScreenControllerPS
function WallScreen:GetDevicePS() return end

function WallScreen:Move() return end

function WallScreen:MoveBack() return end

File: DecompiledGameScripts/cyberpunk/devices/homeAppliances/mediaDevices/tvs/wallScreen.swift

public class WallScreen extends TV {

  private let m_movementPattern: SMovementPattern;

  private let m_factOnFullyOpened: CName;

  private let m_objectMover: ref<ObjectMoverComponent>;

  protected cb func OnRequestComponents(ri: EntityRequestComponentsInterface) -> Bool {
    EntityRequestComponentsInterface.RequestComponent(ri, n"object_mover", n"ObjectMoverComponent", false);
    EntityRequestComponentsInterface.RequestComponent(ri, n"tv", n"PhysicalMeshComponent", false);
    super.OnRequestComponents(ri);
  }

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    this.m_objectMover = EntityResolveComponentsInterface.GetComponent(ri, n"object_mover") as ObjectMoverComponent;
    super.OnTakeControl(ri);
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as WallScreenController;
  }

  private const func GetController() -> ref<WallScreenController> {
    return this.m_controller as WallScreenController;
  }

  public const func GetDevicePS() -> ref<WallScreenControllerPS> {
    return this.GetController().GetPS();
  }

  protected cb func OnMovementFinished(movementStatus: ref<ObjectMoverStatus>) -> Bool {
    if Equals(movementStatus.direction, this.m_movementPattern.direction) && IsNameValid(this.m_factOnFullyOpened) {
      SetFactValue(this.GetGame(), this.m_factOnFullyOpened, 1);
    };
    GameObject.PlaySoundEvent(this, n"dev_doors_hidden_stop");
  }

  protected cb func OnToggleSecureShow(evt: ref<ToggleShow>) -> Bool {
    if this.GetDevicePS().IsShown() {
      this.Move();
    } else {
      this.MoveBack();
    };
  }

  private final func Move() -> Void {
    this.UpdateDeviceState();
  }

  private final func MoveBack() -> Void {
    this.UpdateDeviceState();
  }
}
